src/bmi/bmi_sac.f90

```diff
@@ -894,17 +923,26 @@ function sac_get_int(this, name, dest) result (bmi_status)
     character (len=*), intent(in) :: name
     integer, intent(inout) :: dest(:)
     integer :: bmi_status
-
+        
     select case(name)
-!==================== UPDATE IMPLEMENTATION IF NECESSARY FOR INTEGER VARS =================
 !     case("model__identification_number")
 !        dest = [this%model%id]
 !        bmi_status = BMI_SUCCESS
-    case default
-       dest(:) = -1
-       bmi_status = BMI_FAILURE
-       call write_log("Integer value for variable " // name // " not found!", LOG_LEVEL_WARNING)
-    end select
+
+      case("serialization_state")
+         if(.not.allocated(this%model%serialization_buffer) .or. size(this%model%serialization_buffer) == 0) then
+            call write_log("Serialization not set yet!", LOG_LEVEL_WARNING)
+            bmi_status = BMI_FAILURE
+         else
+            dest = size(this%model%serialization_buffer)
```

Comment:
`dest` needs to actually get the *contents* of `serialization_buffer`, so that the caller can save it to whatever sort of file or other storage.